File: msg-sentiment/src/main/java/dev/number6/sentiment/dagger/ChannelMessagesSentimentComprehensionComponent.java
package dev.number6.sentiment.dagger;


import dev.number6.CommonModule;
import dev.number6.message.ChannelMessagesHandler;
import dagger.Component;

@Component(modules = {
        CommonModule.class,
        ChannelMessagesHandlerModule.class,
        ComprehensionResultsModule.class
})
public interface ChannelMessagesSentimentComprehensionComponent {

    ChannelMessagesHandler getChannelMessagesHandler();
}

File: msg-sentiment/src/main/kotlin/dev/number6/sentiment/dagger/ChannelMessagesSentimentComprehensionComponent.kt
package dev.number6.sentiment.dagger

import dagger.Component
import dev.number6.ComprehendModule
import dev.number6.message.ChannelMessagesHandler

@Component(modules = [ComprehendModule::class, ChannelMessagesHandlerModule::class, ComprehensionResultsModule::class])
interface ChannelMessagesSentimentComprehensionComponent {
    fun getChannelMessagesHandler(): ChannelMessagesHandler
}